Reqnroll/FeatureInfo.cs

```diff
@@ -17,12 +18,18 @@ public class FeatureInfo
         public string Description { get; private set; }
         public CultureInfo Language { get; private set; }
 
+
+        // This holds the cucumber messages at the feature level created by the test class generator; populated when the FeatureStartedEvent is fired
+        public FeatureLevelCucumberMessages FeatureCucumberMessages { get; set; }
+        // This holds the unique identifier for the tracker instance that is being used to generate cucumber messages for this Test Case
+        public string CucumberMessages_PickleId { get; set; }
+
         public FeatureInfo(CultureInfo language, string folderPath, string title, string description, params string[] tags)
-            : this(language, folderPath, title, description, ProgrammingLanguage.CSharp, tags)
+            : this(language, folderPath, title, description, ProgrammingLanguage.CSharp, null, tags)
         {
         }
 
-        public FeatureInfo(CultureInfo language, string folderPath, string title, string description, ProgrammingLanguage programmingLanguage, params string[] tags)
+        public FeatureInfo(CultureInfo language, string folderPath, string title, string description, ProgrammingLanguage programmingLanguage, FeatureLevelCucumberMessages featureLevelCucumberMessages = null, params string[] tags)
```

Comment:
Instead of inserting FeatureLevelCucumberMessages to the one before last, please create a new constructor overload, where the tags are not `params`, and the FeatureLevelCucumberMessages is optional and last.
`public FeatureInfo(CultureInfo language, string folderPath, string title, string description, ProgrammingLanguage programmingLanguage, string[] tags, FeatureLevelCucumberMessages featureLevelCucumberMessages = null)`

Reply:
OK. Where this effects tests that new-up a ```FeatureInfo``` to support a test, which constructor should I have them use? The one with or without ```FeatureLevelCucumberMessages```?

Reply:
> OK. Where this effects tests that new-up a `FeatureInfo` to support a test, which constructor should I have them use? The one with or without `FeatureLevelCucumberMessages`?

Meh. NM. Turned out to be not an issue.

Reply:
Please see this [commit](https://github.com/reqnroll/Reqnroll/pull/233/commits/530f1bea45033c563448f5695342e388d9ca1bda) with the changes to the ```FeatureInfo``` constructor.